[18.0][MIG] stock_vertical_lift: Migration to 18.0

LucasTran380381 opened the pull request:
### Put from receipt
* create receipt to shuttle 
![image](https://github.com/user-attachments/assets/a53f0533-6161-4de9-9970-bf00a7774447)
* Switch shuttle to put mode -> Insert barcode product and tray code -> Save to tray location
![image](https://github.com/user-attachments/assets/037e11e5-3ab4-4483-abc7-cf240159d18b)
* Receipt change state to done
![image](https://github.com/user-attachments/assets/18d996bc-a433-4c29-b3e0-d4e79b029a43)
### Pick from Delivery
* Create Delivery
![image](https://github.com/user-attachments/assets/bb83da0c-0c00-4b4b-846c-817dc582bf22)
* Switch Shuttle to Pick mode -> Insert barcode destination -> Save
![image](https://github.com/user-attachments/assets/aa14ec74-e2dc-48f0-bcf2-847cb6b60222)
* Delivery change state to done
![image](https://github.com/user-attachments/assets/473f1863-016a-4e7a-a4ee-eb09ca4678b8)
### Adjust Inventory
* Set stock.quant allow to adjust
![image](https://github.com/user-attachments/assets/684dc1fa-2311-437a-a610-02fd4d51b5df)
* Input real quantity
![image](https://github.com/user-attachments/assets/11d69ed5-c002-4e32-98be-033665c0cd8d)
* Inventory stock.quant has adjusted
![image](https://github.com/user-attachments/assets/6f24b063-ffec-4cd0-94b2-0f96cc9daf47)

mathieuvatel committed:
[ADD] stock_move_location

Commit:
New module stock_inventory_lockdown

This module is a feature extracted from stock_inventory_location.

Ported to v8 and new API
Adopted latest OCA standards
Added tests

oca-transbot committed:
OCA Transbot updated translations from Transifex

MiquelRForgeFlow committed:
stock_inventory_lockdown

oca-transbot committed:
OCA Transbot updated translations from Transifex

JordiBForgeFlow committed:
refactor the module to perform lockdown validations using constraint
 in stock move instead of checks in quants. This will allow for a more
 robust lockdown, and still make it possible to perform inventory adjustments
 in the locked location. Also resolves an outstanding issue related to the
 previous design not allowing inventory adjustments where negative quants existed.

MiquelRForgeFlow committed:
[MIG] stock_inventory_lockdown: Migration to 10.0

LoisRForgeFlow committed:
[10.0][FIX] stock_inventory_lockdown: locking unneded locations.

oca-transbot committed:
OCA Transbot updated translations from Transifex

LoisRForgeFlow committed:
[11.0][MIG] stock_inventory_lockdown

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

Commit:
[MIG] 11.0 stock_move_location

Commit:
[IMP] refactor to use pickings instead of inventories

Commit:
prevent user from creating lots from the wizard

sebalix committed:
[11.0][MIG] stock_move_location: fixing/improving code + add a roadmap
for further improvements

sebalix committed:
[FIX] stock_move_location: fix the 'group_lines()' method

A recordset object is not reliable enough to use as a key for the
built-in 'sorted' and 'itertools.groupby' functions (sometimes it works,
sometimes not).
Using the ID of the record (here the product ID) can fix the problem, but the
'group_lines()' has been totally rewritten for a simpler implementation without
any use of 'sorted' or 'itertools.groupby' functions to group the wizard lines
by product: an iteration on lines to fill a dictionary does the job.

sergio-teruel committed:
Added translation using Weblate (Spanish)

sergio-teruel committed:
[11.0][IMP] stock_move_location: Allow move products from one location to another directly from selected quants

etobella committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (32 of 32 strings)

Translation: stock-logistics-warehouse-11.0/stock-logistics-warehouse-11.0-stock_move_location
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-11-0/stock-logistics-warehouse-11-0-stock_move_location/es/

sergio-teruel committed:
[11.0][FIX] stock_move_location: Not check availability when do immediate transfer.

Commit:
Add stock_vertical_lift module

osimallen committed:
[12.0][MIG]stock_inventory_lockdown

MiquelRForgeFlow committed:
[FIX] Don't check in any case of inventory adjustments

The virtual adjustment location may be different from the one provided in property_stock_inventory of the product.

OCA-git-bot committed:
stock_inventory_lockdown 12.0.1.0.1

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (5 of 5 strings)

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_inventory_lockdown
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_inventory_lockdown/zh_CN/

Commit:
Add method on location to fetch a tray

* Add vertical_lift_shuttle_id field on stock.location, help to find the
shuttle for a location
* Add StockLocation.fetch_vertical_lift_tray(), that needs to be
implemented in addons to send commands to the hardward to fetch a tray,
and if existing show a cell (laser pointer, ...)
* Add helpers on stock.move.line fetch_vertical_lift_tray_source() and
fetch_vertical_lift_tray_dest() that fetch the tray directly from a move
line's source or destination location

Commit:
Split the shuttle operations in different models/views

Namely, the pick/put/inventory operations are now split in
different models.

Pick and Put share a model and customize their behavior, which is pretty
similar. The inventory operation will have a different view and
different workflow.

This changes will ease a lot the customization of the different
workflows and views.

Commit:
Add default views to re-open the screen views properly

When we refresh the page on the browser when we are using the "screen"
view, odoo loses the information that we want the view to be headless,
fullscreen, etc. so it's displayed pretty badly.  This view is a
work-around: its priority is lower, so it will be picked up by default
on loading, and a button allows to re-open the screen view with the
proper options.

Commit:
Add Put-Away workflow

Commit:
Implement inventory screen

Commit:
Fix 'fetch' buttons on move lines

There is no such action as 'ir.actions.do_nothing', it kinda works,
until you look into the js console and stares at the errors.

There is a nice OCA module that serves this purpose (more or less,
because it reloads the window, this is not an issue).

Commit:
Add button on locations to fetch the tray

wildi1 committed:
Added translation using Weblate (German)

gurneyalex committed:
[IMP] abstract communication with shuttle

Commit:
[MIG] stock_move_location to v12
[FIX] improvement move line generation

Commit:
[12.0][IMP] stock_move_location: Applies put away strategy

Commit:
[12.0][FIX][stock_move_location] Planned transfer: quantity done

Commit:
Add method to refresh a shuttle screen

Example of usage in an odoo shell, when a screen is open:

>>> self.env['vertical.lift.shuttle'].browse(1)._operation_for_mode().operation_descr = 'foo'
>>> self.env['vertical.lift.shuttle'].browse(1)._send_notification_refresh()
>>> env.cr.commit()

Provided the longpolling is correctly configured with a proxy, the
screen should immediately refresh with 'foo' as operation description.

JordiBForgeFlow committed:
This commit introduces changes to stock_move_location:
- The ability to start the wizard from a picking type in the inventory dasbhoard
- A toggle to edit the origin and destination locations.

JordiBForgeFlow committed:
in planned transfers do not set the qty_done in move lines

gurneyalex committed:
[IMP] add a proxy to communicate with the kardex server

LoisRForgeFlow committed:
[11.0] stock_move_location: add small margin between buttons,
done inline because the same margin is defined in  style files
for account dashboard, however I don't want to apply the
account_kanban class here as it would have more effects.

JordiBForgeFlow committed:
display picking type in move location wizard, and get a default value

LoisRForgeFlow committed:
[11.0][IMP] stock_move_location:
* fix tests as per latest changes
* extend test as per latest changes
* add little help in technical fields

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (5 of 5 strings)

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_inventory_lockdown
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_inventory_lockdown/pt_BR/

Commit:
[11.0][FIX] stock_move_location:
* fix test as per migration
* fix inconsistency in reserved quantity for planned transfers

Commit:
[IMP] stock_vertical_lift: black, isort

Commit:
[MIG] stock_vertical_lift{,_kardex}: Migration to 13.0

AdriaGForgeFlow committed:
[IMP] stock_inventory_lockdown: black, isort

AdriaGForgeFlow committed:
[MIG] stock_inventory_lockdown: Migration to 13.0

LoisRForgeFlow committed:
[12.0][ADD] stock_demand_estimate_matrix

oca-travis committed:
[UPD] Update stock_demand_estimate_matrix.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-13.0/stock-logistics-warehouse-13.0-stock_inventory_lockdown
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-13-0/stock-logistics-warehouse-13-0-stock_inventory_lockdown/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
stock_demand_estimate_matrix 12.0.2.0.0

JoanSForgeFlow committed:
[IMP] stock_move_location: black, isort

JoanSForgeFlow committed:
[MIG] stock_move_location: Migration to 13.0

Commit:
Replace custom js by 'target': 'main'

sergio-teruel committed:
[FIX] stock_move_location: Keep product lines when user launch wizard from stock quant view

sergio-teruel committed:
[IMP] stock_move_location: Allow to disable apply putaway strategy

HviorForgeFlow committed:
[FIX] stock_move_location: Error on onchange origin location

Commit:
run pre-commit with new prettiers

ps-tubtim committed:
[IMP] stock_demand_estimate_matrix: black, isort

ps-tubtim committed:
[MIG] stock_demand_estimate_matrix: Migration to 13.0

oca-travis committed:
[UPD] Update stock_demand_estimate_matrix.pot

Commit:
[ADD] stock_picking_fillwithstock

Commit:
[IMP] stock_move_location add posibility to set picking instead of creation

Commit:
[IMP] add posibility to move only not reserved quantity

HviorForgeFlow committed:
[13.0][FIX] Broken pre-commit: E741 ambiguous variable name 'l'

OCA-git-bot committed:
stock_inventory_lockdown 13.0.1.0.1

sergio-teruel committed:
[IMP] stock_move_location: Allow to select any location.

gurneyalex committed:
[FIX] command key must be a string

not a list

Commit:
Fix issues in Pick mode

* The change of destination location was not updated on the screen when
  the barcode was scanned (it was when the "manual barcode wizard" is
  used though)
* We should be able to pick partially available move lines
* prevent to scan a location when no move line is selected or the move
  line has already been set to done

Commit:
Execute the barcode scanner method as side-effect method

Instead of going through the onchange machinery.
The intended usage of onchange methods is to update something on the
screen, without side-effects in the database, then let the user save
the form with the proposed changes.

Weirdly, the barcode scanner event triggers an onchange on the field
`_barcode_scanned`.

It doesn't work well with our use case, as the whole form is read-only
and we only care about having the barcode events doing side-effects on
the backend and displaying back the changes.

This particular onchange will then be executed as a normal method, with
side-effects. However, contrarily to other actions on the form, the
frontend does not reload the view after an onchange, as it relies on the
data returned back in the values. As we cannot know which values may
have been changed in the different implementations (location
destination, state, ...), the onchange returns a read with every field.

Commit:
Rework workflows using a small state machine

The documentation of the state machine is in
VerticalLiftOperationBase._transitions.

Commit:
Add stock_vertical_lift_storage_type

Compatibility module between stock_vertical_lift and stock_storage_type
(in OCA/wms).

In the vertical lift's Putaway screen, when a good is scanned for a putaway, the
user has to scan the tray type of the corresponding size, so an empty place in a
matching tray is found. When we use storage types, we should know what tray is
compatible with the storage type.

Changes with this module:

* The storage types of trays cannot be selected in the locations form, they have
  to be set in the Tray types.
* In the lift put-away screen, when a package has a storage type, the user isn't
  asked to scan a tray type, instead, the putaway of the Package Storage Type is
  applied.

Commit:
Allow changing a putaway tray type after assigned

When the shuttle screen propose a tray based on a tray type and we
are in the 'save' step, where we are supposed to physically putaway
the good and save, we should still be able to change the tray type
to fetch another tray.

Commit:
Extract methods to allow more precise overrides

Commit:
Putaway: show rainbow man when all the lines are putaway

Commit:
Add release (close) of vertical lift trays

* Rename methods that fetch a tray to prevent confusion
* Add methods to release a tray
* The Kardex method to fetch a tray has to send "0" in the carrier and
  carrierNext field
* The pick and inventory screens release the tray only when there is no next
  line, because the release is implicit when we fetch the next line,
  the put screen releases everytime because the operator may take time
  to start the next line and we don't know if they are going to scan a
  next line or not.
* Exiting the screen or switching screen between put/pick/put-away has
  to release the tray as well.

Commit:
Prevent crash if 2 packages have the same name

simahawk committed:
stock_vertical_lift: packaging template more flexible

As the template is not used by JS we can pass full objects to it.
This way we can use any recordset information directly in the template
without having to override the method.

simahawk committed:
stock_vertical_lift: make pkg compute more solid (#16)

* stock_vertical_lift: make pkg compute more solid

Somehow sometimes you can get a move line without product
while computing product packaging in inventory.

Make it more defensive and skip packaging rendering if no product is
there.

simahawk committed:
stock_vertical_lift: change x/y tray order

Commit:
inventory: always fetch tray

The check was means as an optimization: no need to fetch at tray already
open. But "fetch_tray" will not only open the tray, it may also move the
laser on the exact position. So  we should do it for every inventory line.

MiquelRForgeFlow committed:
[UPD] Eficent -> ForgeFlow

Commit:
Improve put-away release in vertical lift

If we have several goods to put in the same tray, it is inefficient to
release (close) the tray between each line if we reopen the same tray.
Release the tray only when the last line is reached.

Commit:
Remove _order override in stock.location

The default order of stock.location is 'complete_name'.
Changing it to "name" can have side effects, and shouldn't be required
for stock_inventory_lockdown.

See https://github.com/OCA/stock-logistics-warehouse/pull/811/files#r512797834

OCA-git-bot committed:
stock_inventory_lockdown 13.0.1.0.2

sergio-teruel committed:
[IMP + FIX] stock_move_location: Display reserved quantity in wizard lines. Reassign reserved quantities for quants moved.

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-13.0/stock-logistics-warehouse-13.0-stock_inventory_lockdown
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-13-0/stock-logistics-warehouse-13-0-stock_inventory_lockdown/

Commit:
[FIX] stock_vertical_lift: handles multiple move lines on pick

If there is two move lines for the same product in the vertical lift
(stored in2 differents trays for instance), the pick scenario was
failing when the user was processing the first line.

To circumvent this, instead of validating directly the move, we put the line
in its own stock move, then we put the stock move in its own transfer and
validate this one.

Methods used to do that have been copied from the `shopfloor` module,
they probably deserves their own module as they are quite generic.

joao-p-marques committed:
[MIG] stock_move_location: Migration to 14.0

TT26454

victoralmau committed:
[FIX] stock_vertical_lift: Error in tests

Commit:
Revert "[13.0][FIX] stock_vertical_lift: Error in tests"

Commit:
[IMP] stock_vertical_lift: Add skip operation to vertical lift shuttles

In the screen for the vertical lift shuttles, accessible through
Inventory > Operations > Vertical Lift Shuttles, a new button has been
added to allow to skip an operation. This button can also be triggered
by scanning the barcode O-BTN.skip.svg that is inside the folder
'images'.

When a skip is done, the next stock.move.line to pick is chosen and
shown to the operator. A skipped move line is added to the end of the
list of pending move lines to be picked, so they will be shown again
in the future as soon as the other move lines have been successfully
processed.

This option was added because, sometimes, the operator can not process
a move line for whatever reason. Right now, the only way of proceeding
is to wait until he/she can effectively process it, which involves a
delay in the operations. With this new skip operation, the operator
can continue processing the rest of move lines.

Commit:
[FIX] stock_vertical_lift: skip button is only for pick operation

The button to skip an operation is only implemented for the pick
operation (not for the put or for the inventory ones) but it was
shown in the screens for all the operations, yielding to a stack
trace when it was pressed from the wrong operationg. The button
has been moved now to the screen for the pick operation, only.

dzungtran89 committed:
[IMP] stock_vertical_lift: black, isort, prettier

dzungtran89 committed:
[MIG] stock_vertical_lift: Migration to 14.0

santostelmo committed:
[13] Fix stock_vertical_lift skip screen

Fix unit tests

fixup! [13] Fix stock_vertical_lift skip screen

Rename variable to current_move_line

fixup! Rename variable to current_move_line

DavidBForgeFlow committed:
[IMP] stock_demand_estimate_matrix: black, isort, prettier

DavidBForgeFlow committed:
[MIG] stock_demand_estimate_matrix: Migration to 14.0

Commit:
[ADD] stock_vertical_lift_empty_tray_check: is the tray empty?

A vertical lift retrieves a tray and places it in front of the
user, and depending on the quantity the user takes from it,
it adapts the pending quantity in the tray. However, because of
errors, it could be that the system thinks the tray is empty
while it is not. With this module, when the system thinks the
tray is empty, while in the step for the release of the tray
the operator is asked explicitly to check if the tray is
actually empty. Depending on his/her answer (yes/no) an inventory
adjustment is created stating the situation.

To activate this optional feature, a new configuration setting
has been added to Inventory > Configuration > Settings, named
'Check Empty Tray'. It is deactivated by default.

Developing decisions:
- The screens shown to the operator are actually wizards, but
  since in the original module (`stock_vertical_lift`) they
  were considered (on the source tree) as views, this has been
  continued here.
- It has been decided, to not change the current workflow of
  the operators, to embed the new check inside the step for
  the 'release'. So, a new screen is shown to ask for the
  visual inspection of whether the tray is empty. In
  order to test this easily, the method `button_release` of
  the module `stock_vertical_lift` has been slightly modified
  so that it always returns. This way we can check easily
  in the unit-tests for the outcome of the intermediate
  screen (i.e. wizard) ─ similarly to how it is done when
  validating a picking that can result in a backorder.

oca-travis committed:
[UPD] Update stock_demand_estimate_matrix.pot

OCA-git-bot committed:
[UPD] README.rst

remi-filament committed:
Translated using Weblate (French)

Currently translated at 100.0% (43 of 43 strings)

Translation: stock-logistics-warehouse-14.0/stock-logistics-warehouse-14.0-stock_move_location
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-14-0/stock-logistics-warehouse-14-0-stock_move_location/fr/

oca-travis committed:
[UPD] Update stock_vertical_lift.pot

hailangvn committed:
[IMP] stock_vertical_lift: increase coverage

OCA-git-bot committed:
stock_vertical_lift 14.0.1.1.0

florentx committed:
[FIX] adapt syntax for act_window lookup

OCA-git-bot committed:
stock_vertical_lift 14.0.1.1.1

BernatPForgeFlow committed:
[IMP] stock_move_location: black, isort, prettier

BernatPForgeFlow committed:
[MIG] stock_move_location: Migration to 15.0

LoisRForgeFlow committed:
[15.0][MIG] stock_demand_estimate_matrix

Commit:
[UPD] Update stock_demand_estimate_matrix.pot

OCA-git-bot committed:
[UPD] README.rst

simahawk committed:
stock_vertical_lift: do not log VERTICAL_LIFT_SECRET

simahawk committed:
stock_vertical_lift: add controller tests

OCA-git-bot committed:
stock_vertical_lift 14.0.1.1.2

Commit:
stock_picking_show_linked: Add to 14.0

oca-travis committed:
Update stock_picking_show_linked.pot

OCA-git-bot committed:
icon.png

OCA-git-bot committed:
README.rst

OCA-git-bot committed:
stock_picking_show_linked 14.0.1.0.1

HviorForgeFlow committed:
stock_demand_estimate_matrix: Allow edit demand estimate tree view

OCA-git-bot committed:
stock_demand_estimate_matrix 15.0.1.1.0

Commit:
[14.0][IMP] stock_move_location: packages support

grindtildeath committed:
stock_move_location: Remove update of assigned moves location

The update of the source location of a move is not a good idea, as
shown in the test case that is modified by this commit and explained
below.

In general, a stock.move will be linked to a stock.picking whose
location_id and location_dest_id are defined by the picking type.
As such, the stock.move will inherit these locations and it is the
assignation of the stock.move that will create a stock.move.line
where the location (in case of outgoing picking) or the destination
location (in case of incoming picking through putaway) will be a
sub location of the respective location defined on the move.

So, if we want to empty a stock.location and move all its content
to another stock.location, we do not want to update the location
of the reserved moves, since it can break the reservation mechanism
of Odoo, as the assignation of a move will always look into the
children locations of the move, and it is not supposed to be updated
by any internal transfer, to what could be a children location.

Then, if internal transfers were planned from the location that is
being emptied, but are supposed to be executed after the location
has been emptied, there is probably an issue with the planning and
the sequence in which both transfers are being executed.
Since the internal transfer being updated in the test case was also
moving from a shelf location to its parent (where technically the goods
are already, since they are in a child location), we should prefer not
breaking the standard workflow of Odoo instead of supporting a corner
case with an internal transfer.

I guess we can revisit my assumption if we have a good reason to update
the stock location, but the one that was tested doesn't seem to be
good enough to justify breaking a standard mechanism.

alexeirivera87 committed:
[FIX] stock_move_location: Bug on onchange origin location

Commit:
[MIG] stock_inventory_lockdown: Migration to 14.0

mymage committed:
Added translation using Weblate (Italian)

victoralmau committed:
[FIX] stock_demand_estimate: Change SavepointCase to TransactionCase

OCA-git-bot committed:
stock_demand_estimate_matrix 15.0.1.1.1

Abranes committed:
[14.0][IMP] stock_move_location: owner support

tupaq committed:
[MIG] stock_move_location: Migration to 16.0

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 7.1% (9 of 126 strings)

Translation: stock-logistics-warehouse-14.0/stock-logistics-warehouse-14.0-stock_vertical_lift
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-14-0/stock-logistics-warehouse-14-0-stock_vertical_lift/it/

Commit:
[MIG] stock_picking_show_linked: migration to 15.0

Commit:
[UPD] Update stock_picking_show_linked.pot

OCA-git-bot committed:
[UPD] README.rst

AungKoKoLin1997 committed:
[FIX] stock_move_location

WangTKurata committed:
Translated using Weblate (Japanese)

Currently translated at 97.6% (42 of 43 strings)

Translation: stock-logistics-warehouse-16.0/stock-logistics-warehouse-16.0-stock_move_location
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-16-0/stock-logistics-warehouse-16-0-stock_move_location/ja/

LoisRForgeFlow committed:
[FIX] stock_demand_estimate_matrix: remove action extension

no need to modify action, once the tree view is set as editable it
will work as itended, no need to remove form from the view modes.

OCA-git-bot committed:
stock_demand_estimate_matrix 15.0.1.2.0

bosd committed:
[MIG] stock_picking_show_linked: Migration to 16.0

kaynnan committed:
[MIG] stock_demand_estimate_matrix: Migration to 16.0

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (48 of 48 strings)

Translation: stock-logistics-warehouse-16.0/stock-logistics-warehouse-16.0-stock_move_location
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-16-0/stock-logistics-warehouse-16-0-stock_move_location/es/

petrus-v committed:
FIX stock_move_common_dest: calling _compute_common_dest_move_ids on new record

AlexPForgeFlow committed:
[MIG] stock_inventory_lockdown: Migration to 16.0

etobella committed:
*: Fix pylint

Commit:
[UPD] Update stock_demand_estimate_matrix.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (41 of 41 strings)

Translation: stock-logistics-warehouse-16.0/stock-logistics-warehouse-16.0-stock_demand_estimate_matrix
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-16-0/stock-logistics-warehouse-16-0-stock_demand_estimate_matrix/it/

yajo committed:
[BUILD] stock_move_location: convert readme fragments to markdown

yajo committed:
[DOCS] stock_move_location: improve usage instructions

@moduon MT-3609

Fixes https://github.com/OCA/stock-logistics-warehouse/issues/1882.

Co-authored-by: Gelo Joga <147538094+Gelojr@users.noreply.github.com>

AungKoKoLin1997 committed:
[IMP] stock_move_location: improve planned transfer

This commit ensures stock move lines with the selected lot_id are in the 'assigned' state when a specific product with a lot_id is chosen in the stock move location for a planned transfer. It resolves the issue where the selected lot_id for a specific product was being discarded and a different lot_id was inadvertently assigned in the move line.

AungKoKoLin1997 committed:
[FIX] stock_move_location: reserverd_quantity is always zero when move location from inventory adjustment

yostashiro committed:
[FIX] stock_move_location: _get_locations_domain()

Before this commit, the wrong company could be used in the domain.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (41 of 41 strings)

Translation: stock-logistics-warehouse-16.0/stock-logistics-warehouse-16.0-stock_demand_estimate_matrix
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-16-0/stock-logistics-warehouse-16-0-stock_demand_estimate_matrix/it/

dreispt committed:
Added translation using Weblate (Portuguese)

AungKoKoLin1997 committed:
[UPD] add new parameters

yassine-rhimi committed:
Translated using Weblate (French)

Currently translated at 97.9% (47 of 48 strings)

Translation: stock-logistics-warehouse-16.0/stock-logistics-warehouse-16.0-stock_move_location
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-16-0/stock-logistics-warehouse-16-0-stock_move_location/fr/

SilvioC2C committed:
[IMP] stock_vertical_lift: filter notifications by screen

If the same user opens multiple shuttles operations on different screens,
notifications for a specific shuttle will be displayed on all of them,
regardless of the shuttle operation that triggered the notification.

This commit allows filtering notifications per screen, so that only
screens displaying shuttle operations related to the current notifications
will display them.

Notifications that are not shuttle-related are not filtered out.

SilvioC2C committed:
[IMP] stock_vertical_lift: add autovacuum for shuttle commands

SilvioC2C committed:
[IMP] stock_vertical_lift: lesser UI improvement

Commit:
[UPD] Update stock_picking_show_linked.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: stock-logistics-warehouse-16.0/stock-logistics-warehouse-16.0-stock_picking_show_linked
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-16-0/stock-logistics-warehouse-16-0-stock_picking_show_linked/it/

pilarvargas-tecnativa committed:
[FIX] stock_move_location: correct method _get_putaway_strategy name

TT48457

AungKoKoLin1997 committed:
[IMP] stock_move_location
This improvement commit ensures the operation type is determined by the original location,
aiming to prevent the misuse of incorrect operation types in scenarios where multiple internal transfer options exist.

AungKoKoLin1997 committed:
[FIX] stock_move_location: fix planned transfer incorrect reserved_uom_qty
This PR fixes stock quant matching in _get_gather_domain by setting strict=True,
ensuring precise matching based on lot, package, and location. This change addresses
issues arising from the previous approach, which could lead to the inclusion of
unnecessary quants, such as those without an owner or lot.

sergio-teruel committed:
[FIX] stock_move_location: The new picking is created with other picking type of selected

hs-oerp committed:
[IMP] stock_picking_show_linked: pre-commit auto fixes

MarinaAForgeFlow committed:
[IMP] stock_move_location: use current active company in multicompany environment

hs-oerp committed:
[17.0][MIG] stock_picking_show_linked : Migration to 17.0

JasminSForgeFlow committed:
[IMP] stock_demand_estimate_matrix: black, isort, prettier

JasminSForgeFlow committed:
[MIG] stock_demand_estimate_matrix: Migration to 17.0

Commit:
[MIG] stock_move_location: Migration to 17.0

Commit:
[IMP] stock_move_location: Improved usage of wizard

Ensure that the picking types available for use are for internal transfers.

Not change the picking type if selected before selecting an origin location of the same location default.

Add the exclusion of reserved quantities as optional, to avoid trying to transfer more than really available but also maintaining the option to transfer all even the reserved.

Add new field to show all the quantity of the line to see the difference of the available and reserved.

lmignon committed:
[IMP] stock_move_location: 'Move from location' button also on outgoring picking type

lmignon committed:
[IMP] stock_move_location: Improve UI and add hooks

The sql query is replaced by the read_group method from the orm. This
allows the use of odoo domain and the definition of a new method that can
be overriden to get the domain to retrieve the quants.  Extract the
initialization logic for the stock_move_location_line_ids field into a
dedicated method to allow to be called wy others addons.

Improve the wizard form:

 * Toggle fields are now properly displayed
 * Add a placeholder group to be used by specialized addon to add additional filters

lmignon committed:
[IMP] stock_move_location: speedup tests

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (48 of 48 strings)

Translation: stock-logistics-warehouse-16.0/stock-logistics-warehouse-16.0-stock_move_location
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-16-0/stock-logistics-warehouse-16-0-stock_move_location/it/

JoanSForgeFlow committed:
[IMP] stock_inventory_lockdown: refactor in order to block movement in stock.move.line confirmation

JoanSForgeFlow committed:
[IMP] stock_inventory_lockdown: do not block movements of parent location if child bin location is being reviwed

Commit:
[UPD] Update stock_demand_estimate_matrix.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-17.0/stock-logistics-warehouse-17.0-stock_demand_estimate_matrix
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-17-0/stock-logistics-warehouse-17-0-stock_demand_estimate_matrix/

ArnauCForgeFlow committed:
[IMP] stock_inventory_lockdown: unlink test, clean code

fd-oerp committed:
[IMP] stock_picking_show_linked: pre-commit auto fixes

fd-oerp committed:
[18.0][MIG] stock_picking_show_linked: Migration to 18.0

- Renamed variables for better clarity and consistency.
- Removed "Dest. Transfers" button in favor of Odoo's "Next Transfer."
- Renamed "Origin Transfers" button to "Previous Transfer" for consistency.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: stock-logistics-warehouse-16.0/stock-logistics-warehouse-16.0-stock_inventory_lockdown
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-16-0/stock-logistics-warehouse-16-0-stock_inventory_lockdown/it/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (5 of 5 strings)

Translation: stock-logistics-warehouse-16.0/stock-logistics-warehouse-16.0-stock_inventory_lockdown
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-16-0/stock-logistics-warehouse-16-0-stock_inventory_lockdown/pt_BR/

Commit:
[MIG] stock_vertical_lift: Migration to 18.0

Commit:
[IMP] stock_vertical_lift: pre-commit auto fixes

chienandalu committed:
[FIX] stock_move_location: prevent putaway rules when we don't need them

If we disable in the wizard the "Apply putaway rules" toggle we wan't to
prevent those to be applied by any automatic logic. Simply injecting the
avoid_putaway_rules context the core does so ignoring them.

TT52561

chaule97 commented on stock_vertical_lift/models/stock_location.py:
```diff
@@ -142,7 +147,10 @@ def fetch_vertical_lift_tray(self, cell_location=None):
+                _(
+                    "Cannot fetch a vertical lift tray on location %s",
+                    self.name,
+                )
```
```suggestion
              self.env._(
                    "Cannot fetch a vertical lift tray on location %s",
                    self.name,
                )
```

chaule97 commented on stock_vertical_lift/models/vertical_lift_operation_base.py:
```diff
@@ -393,28 +388,18 @@ class VerticalLiftOperationTransfer(models.AbstractModel):
         string="Packaging", compute="_compute_product_packagings"
     )
-    qty_done = fields.Float(related="current_move_line_id.qty_done", readonly=True)
+    # qty_done = fields.Float(related="current_move_line_id.qty_done", readonly=True)
```
a question: why don't you delete it?

chaule97 commented on stock_vertical_lift/models/vertical_lift_operation_base.py:
```diff
@@ -497,7 +486,7 @@ def count_move_lines_to_do_all(self):
         line = self.current_move_line_id
         if line.state in ("assigned", "partially_available"):
-            line.qty_done = line.product_qty
+            # line.qty_done = line.product_qty
```
above question

chaule97 commented on stock_vertical_lift/models/vertical_lift_operation_inventory.py:
```diff
@@ -78,9 +78,10 @@ def _transitions(self):
-    )
+    # current_inventory_line_id = fields.Many2one(
+    #     comodel_name="stock.inventory.line", readonly=True
+    # )
```
same question

chaule97 commented on stock_vertical_lift/static/src/scss/vertical_lift.scss:
```diff
@@ -127,9 +127,9 @@
+    //    padding: 1em;
+    //    font-size: 2em;
+    //    text-transform: uppercase;
+    //}
```
I think we should delete

chaule97 commented on stock_vertical_lift/views/vertical_lift_operation_pick_views.xml:
```diff
@@ -29,7 +29,7 @@
                     class="btn-danger"
                     barcode_trigger="skip"
-                    attrs="{'invisible': [('state', '=', 'noop')]}"
+                    invisible="state != 'noop'"
```
in old code: `attrs="{'invisible': [('state', '=', 'noop')]}"`. is there any change?

LucasTran380381 replied:
just migration syntax suitable for v18

chaule97 replied:
I mean: you change `=` to `!=`

chaule97 replied:
I think it should have be: `invisible="state == 'noop'"`

nilshamerlinck commented on stock_vertical_lift/static/src/scss/vertical_lift.scss:
```diff
@@ -0,0 +1,135 @@
+        }
+    }
+
+    //footer .btn {
+    //    padding: 1em;
+    //    font-size: 2em;
+    //    text-transform: uppercase;
+    //}
```
leftover?

chaule97 commented on stock_vertical_lift/models/vertical_lift_operation_base.py:
```diff
@@ -0,0 +1,504 @@
+        quants = self.env["stock.quant"].search(
+            [("location_id", "=", location.id), ("product_id", "=", product.id)]
+        )
+        return sum(quants.mapped("quantity"))
```
```suggestion
    def _get_tray_qty(self, product, location):
        domain = [("location_id", "=", location.id), ("product_id", "=", product.id)]
        grouped_data = self.env["stock.quant"].read_group(
            domain=domain,
            fields=["quantity"],
            groupby=[]
        )
        return grouped_data[0]["quantity"] if grouped_data else 0.0
```

yankinmax committed:
[IMP] stock_move_location: pre-commit auto fixes

yankinmax committed:
[MIG] stock_move_location: Migration to 18.0

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (40 of 40 strings)

Translation: stock-logistics-warehouse-17.0/stock-logistics-warehouse-17.0-stock_demand_estimate_matrix
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-17-0/stock-logistics-warehouse-17-0-stock_demand_estimate_matrix/it/

nilshamerlinck commented:
up @chaule97

imlopes committed:
[18.0][ADD] stock_picking_stage

Co-authored-by: Ricardoalso <112623095+Ricardoalso@users.noreply.github.com>

arnaudlayec committed:
Translated using Weblate (French)

Currently translated at 100.0% (5 of 5 strings)

Translation: stock-logistics-warehouse-16.0/stock-logistics-warehouse-16.0-stock_inventory_lockdown
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-16-0/stock-logistics-warehouse-16-0-stock_inventory_lockdown/fr/

carlos-lopez-tecnativa committed:
[MIG] stock_inventory_discrepancy: Migration to version 18.0

OCA-git-bot committed:
Merge PR #2290 into 18.0

Signed-off-by jbaudoux

Commit:
[UPD] Update stock_picking_completion_info.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2294 into 18.0

Signed-off-by jbaudoux

Commit:
[UPD] Update stock_packaging_calculator.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

ThiagoMForgeFlow committed:
[IMP] stock_demand_estimate_matrix: pre-commit auto fixes

ThiagoMForgeFlow committed:
[MIG] stock_demand_estimate_matrix: Migration to 18.0

OCA-git-bot committed:
Merge PR #2298 into 18.0

Signed-off-by LoisRForgeFlow

Commit:
[UPD] Update stock_demand_estimate_matrix.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2283 into 18.0

Signed-off-by LoisRForgeFlow

Commit:
[UPD] Update stock_route_location_source.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2295 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update stock_inventory_discrepancy.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: stock-logistics-warehouse-18.0/stock-logistics-warehouse-18.0-stock_route_location_source
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-18-0/stock-logistics-warehouse-18-0-stock_route_location_source/it/

natuan9 committed:
[IMP] stock_inventory_lockdown: pre-commit auto fixes

natuan9 committed:
[MIG] stock_inventory_lockdown: Migration to 18.0

Commit:
[UPD] Update stock_picking_show_linked.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2307 into 18.0

Signed-off-by LoisRForgeFlow

Commit:
[UPD] Update stock_picking_show_linked.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2305 into 18.0

Signed-off-by rousseldenis

Commit:
[UPD] Update stock_inventory_lockdown.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2310 into 18.0

Signed-off-by sebalix

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2276 into 18.0

Signed-off-by jbaudoux

Commit:
[UPD] Update stock_picking_stage.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

OCA-git-bot committed:
Merge PR #2257 into 18.0

Signed-off-by simahawk

Commit:
[UPD] Update stock_location_tray.pot

OCA-git-bot committed:
[BOT] post-merge updates

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 100.0% (3 of 3 strings)

Translation: stock-logistics-warehouse-18.0/stock-logistics-warehouse-18.0-stock_picking_product_assortment
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-18-0/stock-logistics-warehouse-18-0-stock_picking_product_assortment/nl/

CarlosRoca13 committed:
[FIX] stock_move_location: Avoid error with non uom when the group is not set

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: stock-logistics-warehouse-18.0/stock-logistics-warehouse-18.0-stock_picking_stage
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-18-0/stock-logistics-warehouse-18-0-stock_picking_stage/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (54 of 54 strings)

Translation: stock-logistics-warehouse-18.0/stock-logistics-warehouse-18.0-stock_location_tray
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-18-0/stock-logistics-warehouse-18-0-stock_location_tray/it/

OCA-git-bot committed:
Merge PR #2316 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update stock_move_location.pot

OCA-git-bot committed:
[BOT] post-merge updates
